src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecdsa.c

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_ecdsa.h"
+#include "pal_utilities.h"
+#include "pal_bignum.h"
+
+static jobject
+CryptoNative_GetEsDsaSignatureObject()
+{
+    JNIEnv* env = GetJNIEnv();
+    jstring algorithmName = JSTRING("SHA1withECDSA");
+    jobject signatureObject = (*env)->CallStaticObjectMethod(env, g_SignatureClass, g_SignatureGetInstance, algorithmName);
+    (*env)->DeleteLocalRef(algorithmName);
+    if (CheckJNIExceptions()) return NULL;
```

Comment:
Are we not printing the exception here? Does this indicate a sort of catastrophic failure?

Reply:
The `CheckJNIExceptions()` method prints out the exceptions to the Android logs.

Reply:
Okay. So `NULL` is okay here then, but we immediately return in all places where `NULL` is returned but don't record in that scenario. It seems to me that this code should be as boiler plate as possible and this function handles errors but does the caller so I am just trying to understand how reconciles ownership of when to report and when not to. Also, I think we can pass in `env` right?

Reply:
I like having the `CheckJNIExceptions()` call as close to the throwing Java method call as possible for clarity. Since this method calls the Java `Signature.getInstance` method, it checks if it threw an exception.

The callers of this method don't need to check if it threw an exception since we already checked that.

I think a good way to encapsulate the model for JNI exception checking in the PAL is, "a JNI exception will never be left pending across a PAL function return." Or at least that's my mental model of it.

Reply:
I think what I'm getting at is this function seems like 3 lines in 2 places and can be removed entirely.  Just my opinion. Up to you.

Reply:
Yeah that's true. I'll think about removing it. We might want something similar for RSA signing, so I might generalize it for that.